### PR TITLE
remove last references to v0.18 in docs and code

### PR DESCRIPTION
**Pull request by topper-123**
- [x] xref #27463

Somehow all the changes didn't get added to #27463, so here's a follow-up.

I've also removed a reference to 0.17 in style.ipynb which also makes the styler api non-provisional.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -3604,11 +3604,6 @@ def swaplevel(self, i=-2, j=-1, copy=True):
         -------
         Series
             Series with levels swapped in MultiIndex.
-
-        .. versionchanged:: 0.18.1
-
-           The indexes ``i`` and ``j`` are now optional, and default to
-           the two innermost levels of the index.
```
Comment (simonjayhawkins) [inline pandas/core/series.py]: would it make sense to add these two lines to the parameters section?

File: doc/source/user_guide/text.rst [hunk 2]
```diff
@@ -364,16 +364,6 @@ Extracting substrings
...
 
-.. warning::
-
-   In version 0.18.0, ``extract`` gained the ``expand`` argument. When
-   ``expand=False`` it returns a ``Series``, ``Index``, or
-   ``DataFrame``, depending on the subject and regular expression
-   pattern (same behavior as pre-0.18.0). When ``expand=True`` it
-   always returns a ``DataFrame``, which is more consistent and less
-   confusing from the perspective of a user. ``expand=True`` is the
-   default since version 0.23.0.
```
Comment (simonjayhawkins) [inline doc/source/user_guide/text.rst]: should we keep a warning just to say that the default changed in 0.23?

**Review by simonjayhawkins — COMMENTED** [review]
lgtm. just a couple of suggestions.

File: doc/source/user_guide/text.rst (hunk 2 above)
Comment (topper-123) [reply doc/source/user_guide/text.rst]: Yeah, I've changed it back, but just removed the references to 0.18.

File: pandas/core/series.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/series.py]: I've made a update.

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**